test-data/unit/check-based-union-join.test

```diff
@@ -0,0 +1,62 @@
+[case testSimpleUnionAtJoin]
+a = [1, ""]
+reveal_type(a)  # N: Revealed type is "list[int | str]"
+
+
+[case testSubtype]
+class A: pass
+class B(A): pass
+
+reveal_type([A(), B()])  # N: Revealed type is "list[__main__.A]"
+
+
+[case testConditionalExpression]
+class A: pass
+class B: pass
+
+res = A() if bool() else B()
```

Comment:
is it worth also having a test for nested conditions like `A() if bool() else B() if bool2() else C()` or whatever the syntax is